logger unit tests

Keyrxng opened the pull request:
Resolves #5
Requires #4

Keyrxng commented:
@gentlementlegen added some tests covering things. I've also decided to remove `util` and just go with `JSON.stringify`. 

The difference I could see writing these simple tests was not significant enough to remain using it when it may be a problem down the line. Essentially it was including things like array length and such but I don't feel this is awfully useful information when weighing the pros and cons.

0x4007 reviewed (approved):
Looks fine. You can merge if @gentlementlegen doesn't get around to this soon.

gentlementlegen commented on .github/workflows/jest-testing.yml:
Has to be updated to avoid comment spam. Refer to https://github.com/ubiquity/ts-template/blob/development/.github/workflows/jest-testing.yml

gentlementlegen commented on jest.config.ts:
See https://github.com/ubiquity/ts-template/blob/development/jest.config.json (also need to add the related package)

gentlementlegen commented on package.json:
```diff
@@ -90,5 +95,6 @@
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha1.ac34549e6aa8e7ead463a7407e1c7390f61a6610"
```
Do we need the sha here?

Keyrxng replied:
After updating my env if this field is not defined then it will be defined automatically by yarn during install and it appends the sha itself. So because of this I left it as is but I don't think it's strictly necessary is it, I will remove it as it's not present in repos using V4 either.

gentlementlegen replied:
I am not sure about the proper course to chose as we do not use yarn 4 extensively in our repositories. As long as it works and that it is supported I am fine with it, but I think it shall work with only `yarn@1.22.22`

gentlementlegen commented on src/logs.ts:
```diff
@@ -0,0 +1,225 @@
+
+    const stackLines = new Error().stack?.split("\n") || [];
+    if (stackLines.length > 3) {
+      const callerLine = stackLines[3]; // .replace(process.cwd(), "");
```
Remove comment probably

gentlementlegen commented on src/logs.ts:
```diff
@@ -0,0 +1,225 @@
+      metadata = {};
+    }
+    if (typeof metadata == "string" || typeof metadata == "number") {
+      // TODO: think i need to support every data type
```
Can this TODO be solved beforehand?

Keyrxng replied:
Well it's not required to handle any other data types really. `Metadata` has `[key: string]: unknown;` so it can already process every data type so long as it's a property of metadata which is cleaner than allowing and having to process others isn't it?

So I think just a clean up here is needed because `_addDiagnosticInformation()` should only really be passed a `Metadata` object or undefined as that's what the public log functions accept as input.

This way invoking logs should just be `.fatal("xyz error", { error: e, a: [], b: {}, ...}`

gentlementlegen commented on src/logs.ts:
```diff
@@ -0,0 +1,225 @@
+      fatal: "-", // - text in red
+      ok: "+", // + text in green
+      error: "!", // ! text in orange
+      // info: "#", // # text in gray
```
Why all the comments?

Keyrxng replied:
These are comments from the original I thought best to leave them but I will do a thorough clean up of those that are not needed

gentlementlegen commented on src/utils.ts:
```diff
@@ -0,0 +1,33 @@
+// eslint-disable-next-line no-control-regex
```
Why is this required? Also this file doesn't seem to be properly formatted.

Keyrxng replied:
https://stackoverflow.com/questions/49743842/javascript-unexpected-control-characters-in-regular-expression

It's needed because of the ansi escaping and you are right I must not have run format

gentlementlegen replied:
I was referring to the eslint rule disable? Does this regex cause any issue?

Keyrxng replied:
No issues, it does the job it's supposed to do 

> Control characters are special, invisible characters in the ASCII range 0-31. These characters are rarely used in JavaScript strings so a regular expression containing elements that explicitly match these characters is most likely a mistake.

From the eslint docs, but in this instance it's not a mistake

Keyrxng commented:
@gentlementlegen I've cleaned things up, bumped ts and the plugins being used and removed those that were not and updated the wf 

Seems that no dependencies are required according to Knip and from running the tests after everything seems to be working so I've removed those too.

Removed what comments were not needed, the code is pretty straight forward so not many are needed but I've left some

gentlementlegen commented on .github/workflows/release-please.yml:
```diff
@@ -0,0 +1,26 @@
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
```
This package has been deprecated, if you can please update it to `googleapis`

Keyrxng replied:
https://github.com/googleapis/release-please-action?tab=readme-ov-file#basic-configuration

The readme includes a comment about setting up a PAT token as the secret
